R/teal_data-get_code.R

```diff
@@ -103,23 +105,26 @@
 #' @aliases get_code,teal_data-method
 #'
 #' @export
-setMethod("get_code", signature = "teal_data", definition = function(object, deparse = TRUE, datanames = NULL, ...) {
-  checkmate::assert_character(datanames, min.len = 1L, null.ok = TRUE)
-  checkmate::assert_flag(deparse)
-
-  code <- if (!is.null(datanames)) {
-    get_code_dependency(object@code, datanames, ...)
-  } else {
-    object@code
-  }
+setMethod("get_code",
+  signature = "teal_data",
+  definition = function(object, deparse = TRUE, names = NULL, datanames = lifecycle::deprecated(), ...) {
+    if (lifecycle::is_present(datanames)) {
+      lifecycle::deprecate_warn(
+        when = "0.6.1",
+        what = "teal.data::get_code(datanames)",
+        with = "teal.code::get_code(names)",
+        always = TRUE
+      )
+      names <- datanames
+    }
 
-  if (deparse) {
-    if (length(code) == 0) {
-      code
-    } else {
-      paste(code, collapse = "\n")
+    if (!is.null(names) && lifecycle::is_present(datanames)) {
```

Comment:
have you seen [`rlang::check_exclusive()`](https://rlang.r-lib.org/reference/check_exclusive.html)?

Reply:
that would be nice

Reply:
What about this comment @m7pr. Is it resolved?